src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -37,12 +38,104 @@
 using namespace std::string_literals;            // NOLINT
 using namespace opossum::expression_functional;  // NOLINT
 
+namespace {
+
+using namespace opossum;  // NOLINT
+
+template <typename Functor>
+void resolve_binary_predicate_evaluator(const PredicateCondition predicate_condition, const Functor functor) {
+  /**
+   * Instantiate @param functor for each PredicateCondition
+   */
+
+  // clang-format off
+  switch (predicate_condition) {
+    case PredicateCondition::Equals:            functor(boost::hana::type<EqualsEvaluator>{});            break;
+    case PredicateCondition::NotEquals:         functor(boost::hana::type<NotEqualsEvaluator>{});         break;
+    case PredicateCondition::LessThan:          functor(boost::hana::type<LessThanEvaluator >{});         break;
+    case PredicateCondition::LessThanEquals:    functor(boost::hana::type<LessThanEqualsEvaluator>{});    break;
+    case PredicateCondition::GreaterThan:       functor(boost::hana::type<GreaterThanEvaluator>{});       break;
+    case PredicateCondition::GreaterThanEquals: functor(boost::hana::type<GreaterThanEqualsEvaluator>{}); break;
+
+    default:
+      Fail("PredicateCondition should be handled in different function");
+  }
+  // clang-format on
+}
+
+std::shared_ptr<AbstractExpression> rewrite_between_expression(const AbstractExpression& expression) {
+  // `a BETWEEN b AND c` is evaluated by transforming it to `a >= b AND a <= c` instead of evaluating it with a
+  // dedicated algorithm. This is because three expression data types (from three arguments) generate many type
```

Comment:
> I simplified the comment so that it doesn't imply universality anymore.

That's all I'm asking for. "`a BETWEEN b AND c` is evaluated by [...]" implies that this is always done this way.